### ISSUE TITLE
[Feature Request] Hotkey/Button integration with Quest Helper

### ISSUE THREAD

**Issue by iPathos** [Skretzo/shortest-path#67]
I've only recently started using the plugin but it has been nothing less than a godsend for my traversal of the game. I have noticed that it's somewhat clunky to readjust the path target per quest stage - I am unsure of the ease of implementation of this suggestion or if it's even doable, but if there were a more streamlined way to set your path target to the current [Quest Helper](https://github.com/Zoinkwiz/quest-helper) objective/target I feel like it would be an awesome addition.

Keep doing what you're doing and thanks again. :)

**Comment by Skretzo** [comment]
Integration with other plugins has been worked on in #48, but I would like to see https://github.com/runelite/runelite/pull/16863 in the core RuneLite client being implemented before continuing the development of cross plugin communication. I know that the Quest Helper developer is interested in this, so it will hopefully happen eventually.

**Comment by Zoinkwiz** [comment]
Cross-plugin comunication now exists, so if this is implemented in shortest-path I can look to interface with it in quest helper.

**Comment by Skretzo** [comment]
> Cross-plugin comunication now exists, so if this is implemented in shortest-path I can look to interface with it in quest helper.

Before I implement cross-plugin communication I want to streamline the collision map generation and also implement weekly automatic collision map updating as the player is running around in-game. Otherwise the plugin is not very useful when new content is released, which I imagine is especially important for the quest helper and new quest releases. I also have a long way to go when it comes to including all transportation methods available in the game, and I am yet to properly treat varbits and inventory items. So I sadly cannot promise that this will happen soon.

**Comment by ctdgunner** [comment]
TL;DR I vote for the sooner option :)

Here's why I say that:
Could it be something that is off by default if you don't think it would be perfect? I would much rather have the ability to use this feature than not have it at all. I also do think there are many more players doing older quest than the new ones. Plenty of players are just playing casually and doing older quests as they just started playing recently. I have used this with quest helper just going to the map every objective and setting to target, it would be nice if it had an option where it just did that automatically. I haven't had a single issue doing this on literally every quest. I think its safe to say the plugin is good enough to add this. especially if you have to go into the config to enable it.

**Comment by iPathos** [comment]
These are some promising updates to hear about! As stated by ctdgunner, it would probably be for the best if such an option (if/when implemented) either wasn't default-on or required a manual bind to activate.

Per Skretzo's comment, I'm likely not familiar enough with coding to offer direct help, regarding transportation methods and the like I can at least offer to cross reference any lists to the best of my ability.

**Comment by Skretzo** [comment]
I have now added support for cross-plugin communication via the [`PluginMessage`](https://github.com/runelite/runelite/blob/master/runelite-client/src/main/java/net/runelite/client/events/PluginMessage.java) event in Shortest Path version 1.15. It has been submitted for review and will probably be live in the client in a few days or 1 to 2 weeks. This means that other plugins like the Quest Helper can now start implementing the use of the Shortest Path pathfinding on their end. Currently the Shortest Path plugin uses the current settings in the plugin to decide which teleports and shortcuts to use, but in the future an option to override the plugin settings can be added if needed.

Other plugins can now request to remove or display a path to a given target location via the following:
- `shortestpath` is the namespace that can be used to specify the Shortest Path plugin.
  - `clear` is the name that can be used to remove the currently displayed path.
  - `path` is the name that can be used to display a path.
    - `start` is the Map key that specifies the path start location (defaults to current location of the player if omitted).
    - `target` is the Map key that specifies the path end location.

Example of how to request displaying a path:
```java
Map<String, Object> data = new HashMap<>();
data.put("start", new WorldPoint(3164, 3483, 0));
data.put("target", new WorldPoint(2834, 3271, 0));
eventBus.post(new PluginMessage("shortestpath", "path", data));
```
Example of how to request removing a path:
```java
eventBus.post(new PluginMessage("shortestpath", "clear"));

```

**Comment by Zoinkwiz** [comment]
Geat! I know you mentioned about it being hard to ensure that the collision maps are updated for new areas. Is that something I can help you obtain for new quests? Seeing as myself and others will need to go through all these areas for making the helpers, if we can assist with running something to collect the collision data or something I'm happy to do so.

**Comment by swirle13** [comment]
> The remaining problems are new agility shortcuts, doors with key requirements, boats, fairy rings, etc. It is very much appreciated if people help out with this, especially agility shortcuts and doors that usually require in-game testing to figure out.

How can we help with that? I've ran into a handful of areas where a shortcut isn't accounted for, e.g. the short wall hop between Sophanem and the ToA Necropolis and the pathing will stop there when calculating and attempt another solution, much farther away, like the magic carpets.

**Comment by FIrgolitsch** [comment]
I think this is working pretty well these days :)
